src/main/java/com/maybe/maybe/service/OrderItemService.java

```diff
@@ -1,13 +1,115 @@
 package com.maybe.maybe.service;
 
+import com.maybe.maybe.dto.OrderItemDTO;
+import com.maybe.maybe.entity.Order;
+import com.maybe.maybe.entity.OrderItem;
+import com.maybe.maybe.entity.Product;
 import com.maybe.maybe.repository.OrderItemRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
+import javax.persistence.EntityNotFoundException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
 @Service
 public class OrderItemService {
     private OrderItemRepository orderItemRepository;
+    private OrderService orderService;
+    private ProductService productService;
 
-    public OrderItemService(OrderItemRepository orderItemRepository) {
+    public OrderItemService(
+            OrderItemRepository orderItemRepository,
+            OrderService orderService,
+            ProductService productService
+    ) {
         this.orderItemRepository = orderItemRepository;
+        this.orderService = orderService;
+        this.productService = productService;
+    }
+
+    public Page<OrderItem> findAllByOrderId(Long orderId, Pageable pageable) {
+        return orderItemRepository.findAllByOrderId(orderId, pageable);
+    }
+
+    public OrderItem getOrderItemById(Long id) {
+        OrderItem order = orderItemRepository.getOrderItemById(id);
+        if (order == null) {
+            throw new EntityNotFoundException("Order item with id: " + id + " not found");
+        } else {
+            return order;
+        }
+    }
+
+    public OrderItem createFromDTO(OrderItemDTO orderItemDTO, Long orderId) {
+        OrderItem orderItem = new OrderItem();
+        orderItem.setOrder(orderService.getOrderById(orderId));
+
+        return getOrderItemAndUpdateOrderTotal(orderItem, orderItemDTO);
+    }
+
+    public OrderItem updateFromDTO(OrderItem orderItem, OrderItemDTO orderItemDTO) {
+
+        orderItem.setOrder(orderService.getOrderById(orderItemDTO.getOrderId()));
+        return getOrderItemAndUpdateOrderTotal(orderItem, orderItemDTO);
+    }
+
+    private OrderItem getOrderItemAndUpdateOrderTotal(OrderItem orderItem, OrderItemDTO orderItemDTO) {
+        orderItem.setProduct(productService.getProductById(orderItemDTO.getProductId()));
+        orderItem.setQuantity(orderItemDTO.getQuantity());
+
+        Product product = productService.getProductById(orderItemDTO.getProductId());
```

Comment:
If execute Product product - ... berfore ourderItem.setProduct then code will be less